src/inmanta/server/services/compilerservice.py

```diff
@@ -478,30 +478,34 @@ async def is_compiling(self, environment_id: uuid.UUID) -> Apireturn:
             return 200
         return 204
 
-    async def _calculate_recompile_wait(self, compile: data.Compile):
-        last_run = await data.Compile.get_last_run(compile.environment)
+    def _calculate_recompile_wait(
+        self, compile_requested: datetime.datetime, last_compile_completed: datetime.datetime, now: datetime.datetime
+    ):
         wait_time = opt.server_autrecompile_wait.get()
-        if not last_run or wait_time == 0:
+        if wait_time == 0:
             wait: float = 0
         else:
-            assert last_run.completed is not None
-            if last_run.completed >= compile.requested:
-                wait = max(0, wait_time - (last_run.completed - compile.requested).total_seconds())
+            assert last_compile_completed is not None
+            if last_compile_completed >= compile_requested:
+                wait = max(0, wait_time - (last_compile_completed - compile_requested).total_seconds())
```

Comment:
Some (small amount of) time has passed since the last compile has ended. Technically, we've waited for that amount of time already.
```suggestion
                wait = max(0, wait_time - (now - compile_requested).total_seconds())
```